quantmsutils/mzml/mzml_statistics.py

```diff
@@ -66,23 +72,63 @@ def consumeChromatogram(self, chromatogram):
+
+        This method extracts peak information, MS level, retention time, and other
+        relevant attributes from the spectrum. It organizes the data into a dictionary
+        and appends it to the batch data. If the MS level is 2 and `id_only` is True,
+        it also extracts precursor information and appends it to `psm_parts`.
+        When the batch data reaches the specified batch size, it triggers a batch write.
+
+        Parameters
+        ----------
+        spectrum : pyopenms.MSSpectrum
+            The spectrum object to process.
         """
 
-        peaks = spectrum.get_peaks()
-        mz_array, intensity_array = peaks[0], peaks[1]
-        peak_per_ms = len(mz_array)
-        base_peak_intensity = float(np.max(intensity_array)) if peak_per_ms > 0 else None
-        total_intensity = float(np.sum(intensity_array)) if peak_per_ms > 0 else None
-        ms_level = spectrum.getMSLevel()
-        rt = spectrum.getRT()
+        peaks = spectrum.get_peaks()  # Get the peaks for the spec spectrum
+        mz_array, intensity_array = (
+            peaks[0],
+            peaks[1],
+        )  # Split the peaks into mz and intensity arrays
+        peak_per_ms = len(mz_array)  # Get the number of peaks in the spectrum
+        base_peak_intensity = (
+            float(np.max(intensity_array)) if peak_per_ms > 0 else None
+        )  # Get the maximum intensity
+        total_intensity = (
+            float(np.sum(intensity_array)) if peak_per_ms > 0 else None
+        )  # Get the total intensity # TODO: Review by @timo and @julianus
+        ms_level = spectrum.getMSLevel()  # Get the MS level of the spectrum
+        rt = spectrum.getRT()  # Get the retention time of the spectrum
 
         if ms_level == 2:
-            precursor = spectrum.getPrecursors()[0]
-            charge_state = precursor.getCharge()
-            exp_mz = precursor.getMZ()
+            precursor = spectrum.getPrecursors()[0]  # Get the first precursor
+            charge_state = precursor.getCharge()  # Charge of first precursor
+            exp_mz = precursor.getMZ()  # Experimental mass to charge ratio of first precursor
+            intensity = precursor.getIntensity()  # Intensity of first precursor
+            precursors = []
+            # capture if more than one precursor
+            index = 0
+            for precursor in spectrum.getPrecursors():
+                logging.info(
+                    "Precursor charge: %s, precursor mz: %s",
+                    precursor.getCharge(),
+                    precursor.getMZ(),
+                )
+                charge_state = precursor.getCharge() # TODO: Review by @timo and @julianus
+                exp_mz = precursor.getMZ() # TODO: Review by @timo and @julianus
+                intensity = precursor.getIntensity() # TODO: Review by @timo and @julianus
+                rt = spectrum.getRT() # TODO: Review by @timo and @julianus
```

Comment:
if you want the rt of the precursor spectrum please use the code I showed you in the google colab